segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -222,10 +219,142 @@ def generate_test_label_map(size_img=(1024, 1024), num_A=100, num_B=100, num_C=1
+    sample_img[0, :, :, 0] = deepcopy(label_mat)
     sample_img_xr = xr.DataArray(sample_img,
                                  coords=[[1], range(size_img[0]), range(size_img[1]), ['segmentation_label']],
                                  dims=['fovs', 'rows', 'cols', 'channels'])
 
     # and return the xarray to pass into calc_dist_matrix, plus the centroid_indices to readjust it
     return sample_img_xr, centroid_indices
+
+
+def generate_two_cell_test_signal_data(size_img, radius, expression,
+                                       pattern, cell_region, cell_center):
+    """
+    This function generates the signal-level test data for a specified channel (nuclear and membrane)
+    based on a given pattern and the marker num we wish to generate data for.
+
+    Args:
+        size_img: the dimensions of the image
+        radius: the radius of the cell
+        marker_num: the marker num of the cell we're looking at
+        expression: whether the channel we're looking at is nuclear or membrane
+        pattern: the specific pattern we wish to generate signal by, for now we'll just assume it's equal
+            to expression (nuclear or membrane)
+        cell_region: the region covered by the cell we're processing
+        cell_center: the center of the cell we're processing
+
+    Returns:
+        channel_signal: a NumPy array with the generated signal according to the parameters provided
+    """
+    cell_center_x = cell_center[0]
+    cell_center_y = cell_center[1]
+
+    # create the signal data array
+    signal_data = np.zeros(size_img)
+
+    # create a probability mask, which will become a set of random values based on a normal distribution
+    # either centered at a peak for nuclear, or around the edge for membrane
+    prob_mask = np.zeros(size_img)
+
+    # get the coordinates of every point in the array
+    prob_mask_rows, prob_mask_cols = np.ogrid[:size_img[0], :size_img[1]]
+
+    # generate an array of numbers valued depending on how close or far away the coordinate is
+    # from the center or the membrane
+    # note that we multiply by -random() for the membrane case because the values
+    # will initially be negative and it will help to make the values consistent
+    # also, we need to center the distribution around the center for nuclear
+    # but around an edge value for membrane
+    if expression == "nuclear":
+        prob_mask = size_img[0] + size_img[1] - np.maximum(np.abs(prob_mask_rows - cell_center_x),
+                                                           np.abs(prob_mask_cols - cell_center_y)) * random()
+        center_value = prob_mask[cell_center_x, cell_center_y]
+        prob_mask = norm.pdf(prob_mask, center_value)
+    else:
+        prob_mask = np.maximum(np.abs(prob_mask_rows - cell_center_x),
+                               np.abs(prob_mask_cols - cell_center_y)) - (size_img[0] + size_img[1]) * -random()
+        edge_value = prob_mask[cell_center_x, cell_center_y - radius]
```

Comment:
Why does membrane  signal involve an offset in y but not in x?

Reply:
I'm trying to center the distribution around a value extracted from the edge of the cell, so I pick one of the many possible edge values of the cell (I could have also done [cell_center_x, cell_center_y + radius] or [cell_center_x - radius, cell_center_y]. I can definitely comment on that since it seems like that's not clear enough as it stands.

Although now that I think about it, it may not be the best way to ensure a membrane-based pattern. Unlike nuclear representation, we may not want to make the assumption that we're most confident about a membrane pattern at the very edge of the cell. For that matter, it may also be unwise to make the assumption that nuclear representation is strongest at the very center. What do you think @ngreenwald ?